src/BoundaryConditions/flux_calculations.jl

```diff
@@ -0,0 +1,70 @@
+# General
+"""
+    get_boundary_flux(model, ::NoFluxCondition, _...)
+
+Construct zero boundary fluxes for a scalar or 2-element vector(x,y)
+"""
+
+function get_boundary_flux(model, ::NoFluxCondition, var::Fields.Field, Ym, Ya)
+    FT = eltype(Ym)
+    flux = Geometry.Cartesian3Vector(FT(0))
+end
+
+# function get_boundary_flux(
+#     model,
+#     ::NoFluxCondition,
+#     ::Geometry.Cartesian12Vector,
+#     Ym,
+#     Ya,
+# )
+#     FT = eltype(Ym)
+#     flux = Geometry.Cartesian12Vector(FT(0), FT(0))
+# end
+
+# Momentum-specific
+"""
+    get_boundary_flux(model, ::DragLawCondition, _...)
+
+Construct non-zero boundary fluxes using the bulk formula and constant or Mohnin-Obukhov-based drag coefficient, Cd
+"""
+
+function get_boundary_flux(model, bc::DragLawCondition, uv, Ym, Ya)
+    FT = eltype(Ym)
+    coefficients = eltype(bc.coefficients) == FT ? bc.coefficients :
+        bc.coefficients(Ym, Ya)
+
+    uv_1 = Operators.getidx(Ym.uv, Operators.Interior(), 1)
```

Comment:
this may not be an issue, but I think the `1` indicates that you get the top interior value of uv_1. Does that mean someone could not use the BC `DragLawCondition` at the bottom of the domain or on another face?

Reply:
I'll echo what @jakebolewski has suggested: let's make a helper function for this. cc @szy21

Reply:
The reason is that this code assumes that `getidx` and `Interior` live in `Operators`. We should minimize the number of places where we assume the structure of ClimaCore-- we're interfacing with it, and we should minimize this interface.

Reply:
Good points. I wonder if this will be addressed by the `ClimaSimulations` interface in a more general way? In the meantime, I'm happy to add a `get_boundary_index` function, or anything else that @jakebolewski would recommend.